src/Core/Content/Product/DataAbstractionLayer/ProductStreamUpdater.php

```diff
@@ -243,6 +256,7 @@ private function getCriteria(array $filters, ?array $ids = null): ?Criteria
 
         $criteria = new Criteria();
         $criteria->addFilter(...$parsed);
+        $criteria->addState(Criteria::STATE_ELASTICSEARCH_AWARE);
```

Comment:
elasticsearch might not have the correct data at this point as the elasticsearch indexing message might not be handled yet

Reply:
you don't know when elasticsearch is updated and has internally updated the search index. it's totally asynchronously. A hacky solution could be do trigger the indexer delayed 5 minutes after updating elasticsearch 🤮

Reply:
you're right, it's probably not worth it and it's too hacky 😞 
Another possible solution to enhance the performance of the indexing process is only get product_streams that are relevant for the change (using product_stream_filter.field)